monkestation/code/modules/projectiles/projectile/special/rocket.dm

```diff
@@ -17,3 +18,21 @@
 	else
 		playsound(src.loc, "sparks", 100, 1)
 	return BULLET_ACT_HIT
+
+/obj/item/projectile/bullet/SRN_rocket/Impact(atom/A)
+	. = ..()
+	if(istype(A, /obj/singularity))
+		var/mob/living/user = firer
+		firer = user
```

Comment:
What is this for?

Reply:
Let me check if the variable change can be removed

Reply:
Yeah it does, my dmb was not updating sometimes while testing stuff. Fixed now